### PR TITLE
Upgrade language server protocol to v3.2.0

### PR DESCRIPTION
**Pull request by kapilmb**

### CODE REVIEW

**Comment by kapilmb** [comment]
This is still work in progress.

File: src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs [hunk 1]
```diff
@@ -73,8 +73,16 @@ public class CompletionItem
 
         public string InsertText { get; set; }
 
+        public Range Range { get; set; }
+
+        public string[] CommitCharacters { get; set; }
```
Comment (daviwil) [inline src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs]: Apparently we've been missing out on some useful stuff here!

File: src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs [hunk 2]
```diff
@@ -83,5 +91,29 @@ public class CompletionItem
...
+        /// <summary>
+        /// Title of the command.
+        /// </summary>
+        /// <returns></returns>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// The identifier of the actual command handler.
+        /// </summary>
+        public string command { get; set; }
```
Comment (daviwil) [inline src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs]: Need to capitalize 2 of the property names

File: src/PowerShellEditorServices.Protocol/LanguageServer/TextDocument.cs [hunk 3]
```diff
@@ -33,47 +77,103 @@ public class TextDocumentPosition : TextDocumentIdentifier
...
     }
 
+    /// <summary>
+    /// The parameters sent in a save text document notification.
+    /// </summary>
     public class DidSaveTextDocumentParams
     {
-        public TextDocumentIdentifier TextDocument { get; set; }
+        /// <summary>
+        /// The document that was closed.
```
Comment (daviwil) [inline src/PowerShellEditorServices.Protocol/LanguageServer/TextDocument.cs]: I think "closed" should be "saved" here

File: test/PowerShellEditorServices.Test.Host/ServerTestsBase.cs [hunk 4]
```diff
@@ -102,19 +102,16 @@ public class ServerTestsBase
             {
                 // Must have read an error?  Keep reading from error stream
                 string errorString = completedRead.Result;
+                Task<string> errorRead = this.serviceProcess.StandardError.ReadToEndAsync();
 
-                while (true)
+                // Lets give the read operation 5 seconds to complete. Ideally, it shouldn't
+                // take that long at all, but just in case...
+                if (errorRead.Wait(5000))
```
Comment (daviwil) [inline test/PowerShellEditorServices.Test.Host/ServerTestsBase.cs]: 👍

**Comment by kapilmb** [comment]
I am just going through all the events and requests again just to make sure if I haven't missed any. I think this is mostly done though. But just making sure. Will ping you as soon as I am done.

**Comment by kapilmb** [comment]
Looks like I missed some types. I think this PR should be ready tomorrow.

**Comment by kapilmb** [comment]
This PR is finally ready to be reviewed!

File: src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs (hunk 2 above)
Comment (kapilmb) [inline src/PowerShellEditorServices.Protocol/LanguageServer/Completion.cs]: fixed

File: src/PowerShellEditorServices.Protocol/LanguageServer/TextDocument.cs (hunk 3 above)
Comment (kapilmb) [inline src/PowerShellEditorServices.Protocol/LanguageServer/TextDocument.cs]: fixed

**Comment by daviwil** [comment]
Excellent work man, thanks a lot!  I just tried out this PR and the VS Code one on my machine, works great!

**Comment by kapilmb** [comment]
Thanks @daviwil !

**Pull request merged**